### PR TITLE
[RFC] fix io; close #1998

### LINKED ISSUE #1998: Something is being printed to stdout rather than the provided IO object
I tried stepping through Plots.jl in the debugger (which prints things to internal buffers in order to suppress excessive output), but the following output ended up in stdout, anyway:
```
Axis:
continuous_values: 0-element Array{Float64,1}
  discrete_map: Dict{Any,Any}()
discrete_values: 0-element Array{Any,1}
       extrema: Plots.Extrema(Inf, -Inf)
          flip: false
foreground_color_axis: :match
foreground_color_border: :match
foreground_color_grid: :match
foreground_color_guide: :match
foreground_color_minor_grid: :match
foreground_color_text: :match
     formatter: :auto
          grid: true
     gridalpha: 0.1
 gridlinewidth: 0.5
     gridstyle: :solid
         guide: ""
guide_position: :auto
guidefontcolor: :match
guidefontfamily: :match
guidefonthalign: :hcenter
guidefontrotation: 0.0
 guidefontsize: 11
guidefontvalign: :vcenter
        letter: :y
          lims: :auto
          link: 0-element Array{Any,1}
     minorgrid: false
minorgridalpha: 0.05
minorgridlinewidth: 0.5
minorgridstyle: :solid
    minorticks: false
        mirror: false
      rotation: 0
         scale: :identity
          show: true
      showaxis: true
tick_direction: :in
 tickfontcolor: :match
tickfontfamily: :match
tickfonthalign: :hcenter
tickfontrotation: 0.0
  tickfontsize: 8
tickfontvalign: :vcenter
         ticks: :auto
     use_minor: false
         widen: true
```

This usually happens because of an io method like `show(io::IO, o) = print(o.something)` rather than `show(io::IO, o) = print(io, o.something)`. Hope the above output might let you identify which object this could be.

### PR DESCRIPTION
**Pull request by mkborregaard**

### CODE REVIEW

**Comment by mkborregaard** [comment]
cc @Keno

File: src/utils.jl [hunk 1]
```diff
@@ -814,24 +814,24 @@ function debugplots(on = true)
...
 
-debugshow(x) = show(x)
-debugshow(x::AbstractArray) = print(summary(x))
+debugshow(io, x) = show(io, x)
+debugshow(io, x::AbstractArray) = print(io, summary(x))
 
-function dumpdict(plotattributes::KW, prefix = "", alwaysshow = false)
+function dumpdict(io::IO, plotattributes::KW, prefix = "", alwaysshow = false)
   _debugMode.on || alwaysshow || return
   println()
```
Comment (Keno) [inline src/utils.jl]: `println(io)`

File: src/utils.jl [hunk 2]
```diff
@@ -814,24 +814,24 @@ function debugplots(on = true)
...
   if prefix != ""
     println(prefix, ":")
```
Comment (Keno) [inline src/utils.jl]: `println(io, prefix`

File: src/utils.jl [hunk 3]
```diff
@@ -814,24 +814,24 @@ function debugplots(on = true)
...
   end
   for k in sort(collect(keys(plotattributes)))
     @printf("%14s: ", k)
-    debugshow(plotattributes[k])
+    debugshow(io, plotattributes[k])
     println()
```
Comment (Keno) [inline src/utils.jl]: `println(io)`

File: src/utils.jl [hunk 4]
```diff
@@ -814,24 +814,24 @@ function debugplots(on = true)
...
   end
   println()
```
Comment (Keno) [inline src/utils.jl]: `println(io)`

**Comment by mkborregaard** [comment]
Jesus, I must have been still sleeping this morning when I did that. Fixed, merging. I hope this fixes it.

**Pull request merged**